### ISSUE TITLE
hildingr - Operator is blocked when sequencer is down on Arbitrum

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-05-Index-judging#321]
hildingr

medium

# Operator is blocked when sequencer is down on Arbitrum

## Summary
When the sequencer is down on Arbitrum state changes can still happen on L2 by passing them from L1 through the Delayed Inbox.

Users can still interact with the Index protocol but due to how Arbitrum address aliasing functions the operator will be blocked from calling onlyOperator().

## Vulnerability Detail

The `msg.sender` of a transaction from the Delayed Inbox is aliased:

```solidity
L2_Alias = L1_Contract_Address + 0x1111000000000000000000000000000000001111
```

All functions with the `onlyOperator()` modifier are therefore blocked when the sequencer is down.

The issue exists for all modifiers that are only callable by specific EOAs. But the operator of the Aave3LeverageStrategyExtension is the main security risk.

## Impact

The operator has roles that are vital for the safety of the protocol. Re-balancing and issuing/redeeming can still be done when the sequencer is down it is therefore important that the operator call the necessary functions to operate the protocol when the sequencer is down. 

`disengage()` is an important safety function that the operator should always have access especially when the protocol is still in accessible to other users. Changing methodology and adding/removing exchanges are also important for the safety of the protocol. 

## Code Snippet

https://github.com/sherlock-audit/2023-05-Index/blob/3190057afd3085143a31746d65045a0d1bacc78c/index-coop-smart-contracts/contracts/manager/BaseManagerV2.sol#L113-L116

## Tool used

Manual Review

## Recommendation

Change the `onlyOperator()` to check if the address is the aliased address of the operator.

Duplicate of #262

**Comment by hildingr** [comment]
Escalate for 10 USDC

This is not a duplicate off #262 and should be a separate issue. 

This is not about a check if a sequencer is down but rather a peculiarity on how Arbitrum aliases addresses. On the other L2s the manager is still able to reach all functions when the sequencer is down since the aliasing is not done on EOA's initiating a L1->L2 call.

On Arbitrum this is not the case and the manager/operators are completely blocked from controlling the protocol when the sequencer is down. 

As it stands on the other L2's the manager/operators can still govern over the protocol and use all the available safety features if the sequencer is down. As it stands this is not possible on Arbitrum.

Low probability event where it would be crucial for operator/manager to have access:

Sequencer is down for a prolonged time this could be some kind of attack or a technical issue, couple this with volatility in the market either due to the issues with the sequencer or unrelated. The governance should be able to change the safe parameters of position during such an event. 

The recommended changes in the duplicates actually makes this worse in some cases since repaying debt is completely blocked when the sequencer is down. This is not the case for normal AAVE users which always have the ability to repay loans, this is an important safety feature guaranteed by AAVE.

This can be taken further if the Index Team wishes to have the same safety level as a native AAVE user.

New functionality can be  added to allow the operator to repay debt and de-leverage when the sequencer is down. This is a safety feature available to all AAVE users, AAVE users are never blocked from repaying debt but only from taking out additional loans when the sequencer is down. 

This can be done by allowing a new operator L1Operator to access a new rebalancing feature, this L1Operator is a L1 smart-contract that uses L1 oracle data to initiate a L2 repayment of debt and rebalance when the sequencer is down.

**Comment by 0xffff11** [comment]
Thanks. I think the changes of this happening are slim to non. Sequencer has to be down and at the same time the operator rebalance. Also, I the operator is a multisig, not an EOA, unsure if that makes any difference, but still, I do not see the medium here. @IAm0x52 thoughts?

**Comment by IAm0x52** [comment]
Due to the ripcord function always being available to prevent set token liquidation, low seems more appropriate to me.

**Comment by 0xffff11** [comment]
Thanks for the second opinion. As said above, I agree with low

**Comment by hrishibhat** [comment]
Agree with the Lead Judge and Watson on this being low
@hildingr

**Comment by hildingr** [comment]
I will have to disagree here. The ripcord can not be trusted when the sequencer is down. The price could move in either direction while the Oracle price is stale.

Imagine if the price is moving in a direction that would allow a ripcord pull, the ripcord can not be pulled since the oracle has the stale price. 

On optimism and polygon the operator can call disengage() and stop the protocol from going above the max liquidation ratio. On Arbitrum this is not possible and the position can go way beyond the max LTV without being able to delever. The position could even get liquidated before ripcord or disengage can be called. If  0.95 < HF < 1 the position can be instantly liquidated when the sequencer comes back up since no grace period is given to heavily undercollateralized positions.

**Comment by ckoopmann** [comment]
I don't have a strong opinion on wether this is low or medium so will leave it up to the lead watson / judge to decide.

Since we are not planning to deploy on arbitrum for now we will not act on this issue for now, but will review should we deploy this there in the future.

**Comment by hrishibhat** [comment]
@hildingr 
Additional comment from the Lead watson.
> The key consideration here is that the main intention here is to keep the set token from becoming liquidated. So it doesn't really matter what the oracle is at whether it's stale or completely wrong. As long as the oracle being used by the set token matches AAVE then that's all that matters. So as I've stated, ripcord will always protect the set token so this is low

**Comment by hildingr** [comment]
I disagree, the ripcord will not always protect the position. I will give a concrete example:

Stage 1, the sequencer is up: HF = 1.5 

Stage 2, the sequencer is down: HF is -> 1. ripcord can not be pulled due to incorrect internal LTV.

Stage 3, sequencer still down: HF < 0.95. 

Stage 4, the moment the sequencer comes up: Race condition between instant liquidation and pulling the ripcord. This is because HF < 0.95 and no grace period is given by the AAVE sentinel.

The true LTV always matters even if AAVE and Index use the same stale oracle. We can look at it as a discontinuity in the LTV, when the seq is down it is "unknown" to both protocols but in the instant the sequencer comes back the LTV jumps to the true value. The true LTV can be in very dangerous territory, possibly high enough for instant liquidation.

**Comment by hildingr** [comment]
> @hildingr Additional comment from the Lead watson.
> 
> > The key consideration here is that the main intention here is to keep the set token from becoming liquidated. So it doesn't really matter what the oracle is at whether it's stale or completely wrong. As long as the oracle being used by the set token matches AAVE then that's all that matters. So as I've stated, ripcord will always protect the set token so this is low

I disagree, the ripcord will not always protect the position. I will give a concrete example:

Stage 1, the sequencer is up: HF = 1.5 

Stage 2, the sequencer is down: HF is -> 1. ripcord can not be pulled due to incorrect internal LTV.

Stage 3, sequencer still down: HF < 0.95. 

Stage 4, the moment the sequencer comes up: Race condition between instant liquidation and pulling the ripcord. This is because HF < 0.95 and no grace period is given by the AAVE sentinel.

The true LTV always matters even if AAVE and Index use the same stale oracle. We can look at it as a discontinuity in the LTV, when the seq is down it is "unknown" to both protocols but in the instant the sequencer comes back the LTV jumps to the true value. The true LTV can be in very dangerous territory, possibly high enough for instant liquidation.

**Comment by IAm0x52** [comment]
> Stage 4, the moment the sequencer comes up: Race condition between instant liquidation and pulling the ripcord. This is because HF < 0.95 and no grace period is given by the AAVE sentinel.

This is correct. it would create race conditions under these circumstances

**Comment by hrishibhat** [comment]
Result:
Medium
Unique
Considering this a valid medium based on the above comments

**Comment by sherlock-admin2** [comment]
Escalations have been resolved successfully!

Escalation status:
- [hildingr](https://github.com/sherlock-audit/2023-05-Index-judging/issues/321/#issuecomment-1613336884): accepted
